packages/protocol/contracts/common/UniswapFeeHandlerSeller.sol

```diff
@@ -0,0 +1,158 @@
+    uint256 maxSlippage // as fraction,
+  ) external {
+    require(
+      buyToken == registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID),
+      "Buy token can only be gold token"
+    );
+
+    // An improvement to this function would be to allow the user to pass a path as argument
+    // and if it generates a better outcome that the ones enabled that gets used
+    // and the user gets a reward
+
+    IERC20 goldToken = getGoldToken();
+    // address celoAddress = address(getGoldToken());
+
+    uint256 bestRouterIndex = 0;
+    uint256 bestRouterQuote = 0;
+
+    address[] memory path = new address[](2);
+    // address[] memory thisTokenRouterAddresses = routerAddresses[sellTokenAddress];
+
+    require(routerAddresses[sellTokenAddress].length > 0, "routerAddresses should be non empty");
+
+    // IERC20 token = IERC20(sellTokenAddress);
+    uint256 balanceToBurn = IERC20(sellTokenAddress).balanceOf(address(this));
+
+    for (uint256 i = 0; i < routerAddresses[sellTokenAddress].length; i++) {
+      address poolAddress = routerAddresses[sellTokenAddress][i];
+      IUniswapV2RouterMin router = IUniswapV2RouterMin(poolAddress);
+
+      path[0] = sellTokenAddress;
+      path[1] = address(goldToken);
+
+      // using the second return value becuase it's the last argument
+      // the previous values show how many tokens are exchanged in each path
+      // so the first value would be equivalent to balanceToBurn
+      uint256 wouldGet = router.getAmountsOut(balanceToBurn, path)[1];
+      emit ReceivedQuote(poolAddress, wouldGet);
+      if (wouldGet > bestRouterQuote) {
+        bestRouterQuote = wouldGet;
+        bestRouterIndex = i;
+      }
+    }
+
+    // don't try to exchange on zero quotes
+    if (bestRouterQuote == 0) {
+      return;
+    }
+
+    address bestRouterAddress = routerAddresses[sellTokenAddress][bestRouterIndex];
+    IUniswapV2RouterMin bestRouter = IUniswapV2RouterMin(bestRouterAddress);
+
+    uint256 minAmount = 0;
+    if (maxSlippage != 0) {
+      address pair = IUniswapV2FactoryMin(bestRouter.factory()).getPair(
+        sellTokenAddress,
+        address(goldToken)
+      );
+      minAmount = calculateMinAmount(
+        IERC20(sellTokenAddress).balanceOf(pair),
+        goldToken.balanceOf(pair),
+        amount,
+        maxSlippage
+      );
+    }
+
+    IERC20(sellTokenAddress).approve(bestRouterAddress, balanceToBurn);
+    bestRouter.swapExactTokensForTokens(
+      balanceToBurn,
+      minAmount,
+      path,
+      address(this),
+      block.timestamp + MAX_TIMESTAMP_BLOCK_EXCHANGE
+    );
+
+    goldToken.transfer(msg.sender, goldToken.balanceOf(address(this)));
+    emit RouterUsed(bestRouterAddress);
+    // _sendBackCeloBalance();
+  }
+
+  // function _sendBackCeloBalance() private {
+  //   IERC20 goldToken = getGoldToken();
+  //   // IERC20 stableAsERC20 = IERC20(sellTokenAddress);
+  // }
+
+  // TODO move this to abstract class
+  function calculateMinAmount(
```

Comment:
Since this method is identical in MentoFeeHandlerSeller and UniswapFeeHandlerSeller we could extract it to library

Reply:
will fix after merging